.github/workflows/conformance-gateway-api.yaml

```diff
@@ -124,7 +124,8 @@ jobs:
             --set securityContext.privileged=true \
             --set gatewayAPI.enabled=true
 
-          kubectl wait -n kube-system --for=condition=Ready --all pod --timeout=${{ env.timeout }}
+          kubectl wait -n kube-system --for=condition=Ready -l app.kubernetes.io/part-of=cilium pod --timeout=${{ env.timeout }}
```

Comment:
Good findings :100:.

Depsite adding label selector for cilium only, only coredns will be restarted after cilium installation, so I think this should fix the flake.